src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.KeepAlive.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.WebSockets
+{
+    internal sealed partial class ManagedWebSocket : WebSocket
+    {
+        private bool IsUnsolicitedPongKeepAlive => _keepAlivePingState is null;
+        private static bool IsValidSendState(WebSocketState state) => Array.IndexOf(s_validSendStates, state) != -1;
+        private static bool IsValidReceiveState(WebSocketState state) => Array.IndexOf(s_validReceiveStates, state) != -1;
+
+        private void HeartBeat()
+        {
+            if (IsUnsolicitedPongKeepAlive)
+            {
+                UnsolicitedPongHeartBeat();
+            }
+            else
+            {
+                KeepAlivePingHeartBeat();
+            }
+        }
+
+        private void UnsolicitedPongHeartBeat()
+        {
+            if (NetEventSource.Log.IsEnabled()) NetEventSource.Trace(this);
+
+            Observe(
+                TrySendKeepAliveFrameAsync(MessageOpcode.Pong));
+        }
+
+        private ValueTask TrySendKeepAliveFrameAsync(MessageOpcode opcode, ReadOnlyMemory<byte>? payload = null)
```

Comment:
Why is this `ReadOnlyMemory<byte>? payload = null` rather than `ReadOnlyMemory<byte> payload = default`? i.e. why do we need to distinguish between null and empty?